app/src/server/api/routers/externalModel.router.ts

```diff
@@ -0,0 +1,40 @@
+import { z } from "zod";
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+import { prisma } from "~/server/db";
+import { requireNothing, requireCanModifyProject } from "~/utils/accessControl";
+import {
+  defaultEvaluationModel,
+  findPredefinedEvaluationModel,
+} from "~/utils/externalModels/evaluationModels";
+
+export const externalModelRouter = createTRPCRouter({
+  get: protectedProcedure
+    .input(
+      z.object({
+        id: z.string().optional(),
+        projectId: z.string().optional(),
```

Comment:
Any reason not to make `projectId` required?

Reply:
Same question for `id`

Reply:
Instead of having two different endpoints, one of which gets a default predefined model and another gets a model added by the user, I use a single endpoint with optional parameters. Calling a function without parameters returns a default model (GPT-4-Turbo).

Reply:
I understand that this endpoint is going to be used to return both predefined models and models stored in the database, and that makes sense. Under what circumstances is it better to call this endpoint without any parameters, instead of just referencing the default predefined model?

Reply:
You are right, I will probably change this API while working on the actual implementation of custom external models (which I plan to do tomorrow).
The current idea was to put the logic of getting an instance of a model into a single function that determines which model to return instead of adding additional logic to every place where we get a model.
But yeah, it does not make sense when we have different default models for different places.